corehq/apps/fixtures/management/commands/base.py

```diff
@@ -72,19 +74,25 @@ def delete_rows_only_in_sql(self, diffs_file, logfile, chunk_size):
         sql_ids = DiffDocs(diffs_file, None, chunk_size, MISSING_IN_COUCH)
         id_name = self.sql_class()._migration_couch_id_name
         ignored_count = deleted_count = 0
-        for ids in chunked(sql_ids.iter_doc_ids(), chunk_size, list):
-            to_delete = ids
-            couch_status = couch.view("_all_docs", keys=ids, reduce=False)
-            to_delete = []
-            for res in couch_status:
-                if "value" in res and res["value"].get("deleted"):
-                    to_delete.append(res["id"])
-                else:
-                    ignored_count += 1
-                    print("Unexpected: SQL record not deleted in Couch:", res, file=logfile)
-            print("Removed orphaned SQL rows:", to_delete, file=logfile)
-            self.sql_class().objects.filter(**{f"{id_name}__in": to_delete}).delete()
-            deleted_count += len(to_delete)
+        row_count = len(sql_ids)
+        row_ids = with_progress_bar(sql_ids.iter_doc_ids(), length=row_count, oneline=False)
+        try:
+            for ids in chunked(row_ids, chunk_size, list):
+                couch_status = couch.view("_all_docs", keys=ids, reduce=False)
+                to_delete = []
+                for res in couch_status:
+                    if "value" in res and res["value"].get("deleted"):
+                        to_delete.append(res["id"])
+                    else:
+                        ignored_count += 1
+                        print("Unexpected: SQL record not deleted in Couch:", res, file=logfile)
+                print("Removed orphaned SQL rows:", to_delete, file=logfile)
+                self.sql_class().objects.filter(**{f"{id_name}__in": to_delete}).delete()
+                deleted_count += len(to_delete)
+        except KeyboardInterrupt:
+            traceback.print_exc(file=logfile)
+            del row_ids  # gc with_progress_bar generator, finalize output
```

Comment:
For my own understanding can you say why this is needed

Reply:
Finalizing the `with_progress_bar` generator makes it print its last line of output (elapsed time). If this is not done here it will not happen until the function scope ends, which means the "Aborted" message will be printed before the last line of progress output.